aws/resource_aws_macie2_classification_job.go

```diff
@@ -0,0 +1,887 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/macie2"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/naming"
+)
+
+func resourceAwsMacie2ClassificationJob() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceMacie2ClassificationJobCreate,
+		ReadWithoutTimeout:   resourceMacie2ClassificationJobRead,
+		UpdateWithoutTimeout: resourceMacie2ClassificationJobUpdate,
+		DeleteWithoutTimeout: resourceMacie2ClassificationJobDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"custom_data_identifier_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"schedule_frequency": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"daily_schedule": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"weekly_schedule": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+						"monthly_schedule": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"sampling_percentage": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"name_prefix"},
+			},
+			"name_prefix": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"name"},
+			},
```

Comment:
The length of `name_prefix` has to be shorter than 500, because of the random part. In the example above, note the `500-resource.UniqueIDSuffixLength`